src/backend/distributed/utils/citus_depended_object.c

```diff
@@ -384,3 +411,89 @@ AlterRoleSetStatementContainsAll(Node *node)
 
 	return false;
 }
+
+
+/*
+ * HasDropCommandViolatesOwnership returns true if any object in the given
+ * drop statement violates object ownership.
+ */
+static bool
+HasDropCommandViolatesOwnership(Node *node)
+{
+	List *objects = NIL;
+	Relation relation = NULL;
+	ObjectType objectType = 0;
+
+	if (IsA(node, DropStmt))
+	{
+		DropStmt *dropStmt = castNode(DropStmt, node);
+		objectType = dropStmt->removeType;
+
+		/* DropStmt can accept multiple objects */
+		Node *object = NULL;
+		foreach_ptr(object, dropStmt->objects)
+		{
+			objects = lappend(objects, object);
+		}
+	}
+	else
+	{
+		return false;
+	}
+
+	bool ownershipViolation = false;
+	Node *object = NULL;
+	foreach_ptr(object, objects)
+	{
+		ownershipViolation = ObjectViolatesOwnership(objectType, object, &relation);
```

Comment:
can't we push declaration of `relation` into `ObjectViolatesOwnership`? We don't need the relation here at al?